### PR TITLE
fix(solc): remove default include paths

### PR DESCRIPTION
**Pull request by rkrasiuk**
## Motivation

currently, the default `include_path`s contain the sources, test & script dirs, which lead to the possibility of ambiguous imports like `import 'interface/IContract.sol';` given the project tree
```text
- src
-- Contract.sol
-- interface
--- IContract.sol
```

while it is possible to make the `Graph` resolve such import paths properly, it leads to ambiguity while trying to reconstruct sources for standard json input, since the same node can be imported as `interface/IContract.sol` and `src/interface/IContract.sol`

more info [here](https://docs.soliditylang.org/en/v0.8.13/path-resolution.html#initial-content-of-the-virtual-filesystem)

## Solution

remove default include paths since they are all covered by the base path

## PR Checklist

- [ ] Added Tests
- [ ] Added Documentation
- [ ] Updated the changelog

### CODE REVIEW

File: ethers-solc/src/config.rs [hunk 1]
```diff
@@ -266,6 +266,15 @@ impl ProjectPathsConfig {
                         }
                     }
                 }
+                // if still unable to resolve, attempt to find the absolute import
+                // inside the project paths that's possible due to `--include-dir`
+                for path in [&self.sources, &self.tests, &self.scripts] {
```
Comment (mattsse) [inline ethers-solc/src/config.rs:271]: looping over `include_paths` should be equivalent here, this also has the benefit that we don't add `sources`, `tests` `scripts` as hardcoded include-paths

**Review by mattsse — CHANGES_REQUESTED** [review]

File: ethers-solc/src/config.rs (hunk 1 above)
Comment (rkrasiuk) [reply ethers-solc/src/config.rs:271]: i'm sus, because `include_paths` are a mutable ref here and get modified on every invocation

Comment (mattsse) [reply ethers-solc/src/config.rs:271]: I see, good point,

I think this is fine then, the worst that can happen is that the graph resolves a file but solc can't in which case we still fail as expected.

**Review by mattsse — CHANGES_REQUESTED** [review]
this is the correct fix, 

absolute import paths like `src/...` are already covered by `--base-path <root>` so we don't need to (or shouldn't) add those paths.

but looks like we can remove this function entirely?

**Review by gakonst — CHANGES_REQUESTED** [review]
yeah let's remove the function and remove the usage of this from foundry?

**Comment by rkrasiuk** [comment]
@mattsse @gakonst removed
> remove the usage of this from foundry?
are you talking about disabling custom `include_paths` altogether?
https://github.com/foundry-rs/foundry/blob/433be903a02be0e49c37deacd4f3fe4d71b4c97c/config/src/lib.rs#L155

**Comment by mattsse** [comment]
this we should keep, since this is a valid solc customization, but remove the default paths in `ProjectPathConfig`

**Review by mattsse — APPROVED** [review]

**Review by gakonst — APPROVED** [review]

**Pull request merged**